lib/que/metrics.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require "rack"
+require "benchmark"
+require "prometheus/client"
+require "prometheus/middleware/exporter"
+
+module Que
+  class Metrics
+    def initialize(labels: {})
+      @registry = Prometheus::Client::Registry.new
+
+      @jobs_worked_total = @registry.counter(:que_jobs_worked_total, "Counter for all jobs processed", labels)
+      @jobs_worked_seconds_total = @registry.counter(:que_jobs_worked_seconds_total, "Sum of the time taken processing each job class", labels)
+      @jobs_latency_seconds_total = @registry.counter(:que_jobs_latency_seconds_total, "Sum of time spent by job and priority waiting in queue", labels)
+      @worker_running_seconds_total = @registry.counter(:que_worker_running_seconds_total, "Time since starting to work jobs", labels)
+      @worker_job_acquire_total = @registry.counter(:que_worker_job_acquire_total, "Counter of number of attempts to acquire jobs", labels)
+      @worker_job_acquire_seconds_total = @registry.counter(:que_worker_job_acquire_seconds_total, "Sum of time taken to acquire jobs", labels)
```

Comment:
Can we try to stay within a vaguely reasonable line length, even if it's not enforced yet?